src/https_client.c

```diff
@@ -39,8 +39,14 @@
     } \
   } while(0);
 
+#define GET_PTR(type, var_name, from) \
```

Comment:
Not a huge fan of macros like this. I guess this is a problem with using C.
Optional: To avoid syntactic weirdness, they should probably be wrapped in
```
do { \
  ...
} while (0);
```

Reply:
As I wrote in the pull request description, even I don't like macros, but still a bit better than copy-paste.
For now I would skip the do-while, because I have put everywhere ; after the macro and I hope no one will missuse it.